ptf/saiswitch.py

```diff
@@ -0,0 +1,1642 @@
+        self.assertNotEqual(attr["SAI_SWITCH_ATTR_EGRESS_BUFFER_POOL_NUM"], 0)
+
+        not supported
+        attr = sai_thrift_get_switch_attribute(self.client, ecmp_hash=True)
+        print(attr)
+        self.assertNotEqual(attr["ecmp_hash"], 0)
+        self.assertNotEqual(attr["SAI_SWITCH_ATTR_ECMP_HASH"], 0)
+
+        attr = sai_thrift_get_switch_attribute(self.client, lag_hash=True)
+        print(attr)
+        self.assertNotEqual(attr["lag_hash"], 0)
+        self.assertNotEqual(attr["SAI_SWITCH_ATTR_LAG_HASH"], 0)
+
+        attr = sai_thrift_get_switch_attribute(self.client,
+                                               max_acl_action_count=True)
+        print(attr)
+        self.assertNotEqual(attr["max_acl_action_count"], 0)
+        self.assertNotEqual(attr["SAI_SWITCH_ATTR_MAX_ACL_ACTION_COUNT"], 0)
+        max_acl_action_count = attr["max_acl_action_count"]
+
+        attr = sai_thrift_get_switch_attribute(self.client,
+                                               max_acl_range_count=True)
+        print(attr)
+        self.assertNotEqual(attr["max_acl_range_count"], 0)
+        self.assertNotEqual(attr["SAI_SWITCH_ATTR_MAX_ACL_RANGE_COUNT"], 0)
+
+        s32 = sai_thrift_s32_list_t(int32list=[], count=max_acl_action_count)
+        cap = sai_thrift_acl_capability_t(action_list=s32)
+
+        attr = sai_thrift_get_switch_attribute(self.client,
+                                               acl_capability=cap)
+        print(attr)
+        self.assertNotEqual(attr["acl_capability"], 0)
+        self.assertNotEqual(attr["SAI_SWITCH_ATTR_ACL_CAPABILITY"], 0)
+
+        attr = sai_thrift_get_switch_attribute(self.client,
+                                               max_mirror_session=True)
+        print(attr)
+        self.assertNotEqual(attr["max_mirror_session"], 0)
+        self.assertNotEqual(attr["SAI_SWITCH_ATTR_MAX_MIRROR_SESSION"], 0)
+
+        attr = sai_thrift_get_switch_attribute(self.client,
+                                               default_trap_group=True)
+        print(attr)
+        self.assertNotEqual(attr["default_trap_group"], 0)
+        self.assertNotEqual(attr["SAI_SWITCH_ATTR_DEFAULT_TRAP_GROUP"], 0)
+
+        attr = sai_thrift_get_switch_attribute(self.client,
+                                               acl_stage_ingress=cap)
+        print(attr)
+        self.assertNotEqual(attr["acl_stage_ingress"], 0)
+        self.assertNotEqual(attr["SAI_SWITCH_ATTR_ACL_STAGE_INGRESS"], 0)
+
+        attr = sai_thrift_get_switch_attribute(self.client,
+                                               acl_stage_egress=cap)
+        print(attr)
+        self.assertNotEqual(attr["acl_stage_egress"], 0)
+        self.assertNotEqual(attr["SAI_SWITCH_ATTR_ACL_STAGE_EGRESS"], 0)
+
+
+@group('hw')
+class SwitchHwAttrTest(SaiHelper):
+    '''
+    Switch hardware-related attributes tests
+    '''
+
+    def runTest(self):
+        self.refreshIntervalTest()
+
+    def refreshIntervalTest(self):
```

Comment:
We do not distinguish between hw and model tests. This may be moved to SwitchAttrTest class.